### PR TITLE
Use Wavefront Internal Reporter for RED metrics generation

### PR DESCRIPTION
**Pull request by haosong**
- Remove the logic of Span RED metrics generation from the repo.
- Span RED metrics generation logic is now handled by underlying Wavefront SDKs now.
- Remove the unit test of RED metric generation as well, since this is no longer handled within this repo.

* Pending release of underlying Wavefront SDKs

### CODE REVIEW

File: wavefront-spring-boot-bom/pom.xml [hunk 1]
```diff
@@ -32,7 +32,9 @@
       <dependency>
         <groupId>com.wavefront</groupId>
         <artifactId>wavefront-opentracing-sdk-java</artifactId>
-        <version>1.16</version>
+        <!-- TODO: Remove SNAPSHOT after released -->
+        <!-- <version>1.17</version> -->
+        <version>1.17-SNAPSHOT</version>
```
Comment (snicoll) [inline wavefront-spring-boot-bom/pom.xml:37]: Is the snapshot deployed somewhere? The build is failing as we don't have a repo source for it.

Comment (haosong) [reply wavefront-spring-boot-bom/pom.xml:37]: We don't have SNAPSHOT version deployed anywhere right now, so probably have to wait until we release the official `1.17`

Comment (snicoll) [reply wavefront-spring-boot-bom/pom.xml:37]: @haosong that's a bit annoying. How am I supposed to review if I can't run the build?

Comment (haosong) [reply wavefront-spring-boot-bom/pom.xml:37]: that's why this PR is still in draft. need to build this https://github.com/wavefrontHQ/wavefront-internal-reporter-java/pull/44 locally, and bump other SDK versions to use this snapshot version. so let's just wait until that got released.

Comment (haosong) [reply wavefront-spring-boot-bom/pom.xml]: `wavefront-opentracing-sdk-java:1.17` released, `pom.xml` updated

**Pull request merged**